dbt/adapters/databricks/impl.py

```diff
@@ -234,8 +234,7 @@ def get_relations_without_caching(self, relation: DatabricksRelation) -> Table:
         kwargs = {"relation": relation}
 
         new_rows: List[Tuple[Optional[str], str, str, str]]
-        if relation.database is not None:
-            assert relation.schema is not None
```

Comment:
`assert` does not do anything in production. `pytest` flips a debug switch when running to make the assert active, but you would never want that in production. So this actually doesn't do anything, which can be misleading as it suggests we're testing when we're not.

Reply:
My understanding is that the point of this is to detect if the assertion is violated when debugging/testing.  It doesn't actually do nothing, it just doesn't execute in the deployed code.

Reply:
basically, it is there to verify what we believe to be an invariant.  If it is actually invariant, then we don't need the if clause below.  If clause is definitely safer though.